packages/protocol/contracts/team/TimelockTokenPool.sol

```diff
@@ -168,7 +170,8 @@ contract TimelockTokenPool is EssentialContract {
         Recipient storage r = recipients[recipient];
         uint128 amount;
 
-        for (uint128 i; i < r.grants.length; ++i) {
+        uint256 rGrantsLength = r.grants.length;
```

Comment:
This will reduce gas cost? I thought the optimizer will do such optimization automatically.

Reply:
Many other files can also be optimized similarly. Could you please do that as well?

Reply:
from gas report, I think the optimizer didn't do optimization for this?

Reply:
Sure, will do.